binderhub/repoproviders.py

```diff
@@ -186,16 +187,32 @@ class GitRepoProvider(RepoProvider):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        url, resolved_ref = self.spec.rsplit('/', 1)
+        url, unresolved_ref = self.spec.split('/', 1)
```

Comment:
Why the change from `rsplit` to `split`?

Reply:
See the discussion above: a branch name may contain `/`. Is it possible to obtain url and the ref separately at this stage?

Reply:
The method used above is the only way to get them separately.

In the current implementation the reference is not url encoded like the url, but I think it's better if we DO encode it. Then the `/` is the true divider between the url and ref, instead of the first `/` like now.

Reply:
The discussion is [here](https://github.com/jupyterhub/binderhub/pull/895#discussion_r302481788).